charts/pulsar/templates/broker-service.yaml

```diff
@@ -43,7 +44,17 @@ spec:
   - name: "{{ .Values.tlsPrefix }}pulsarssl"
     port: {{ .Values.broker.ports.pulsarssl }}
   {{- end }}
-  clusterIP: None
+  {{- if and .Values.broker.service .Values.broker.service.clusterIP }}
+  {{- if eq .Values.broker.service.clusterIP "auto" }}
+  clusterIP: ""
+  {{- else if regexMatch "^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])$" .Values.broker.service.clusterIP }}
```

Comment:
I think it's better to not add any validation so that it simplifies things.

Reply:
Cannot find a good motivation. Makes sense to me. 

Addressed by 4d8e283